File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/event/GraphUpdatedEventHandler.java
package jokatu.game.games.uzta.event;

import jokatu.game.Game;
import jokatu.game.event.SpecificEventHandler;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class GraphUpdatedEventHandler extends SpecificEventHandler<GraphUpdatedEvent> {
	@NotNull
	@Override
	protected Class<GraphUpdatedEvent> getEventClass() {
		return GraphUpdatedEvent.class;
	}

	@Override
	protected void handleCastEvent(@NotNull Game<?> game, @NotNull GraphUpdatedEvent event) {
		sender.send("/topic/graph.game." + game.getIdentifier(), event.getGraph());
	}
}

File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/event/GraphUpdatedEventHandler.kt
package jokatu.game.games.uzta.event

import jokatu.components.stomp.Topic
import jokatu.game.Game
import jokatu.game.event.SpecificEventHandler
import ophelia.kotlin.graph.Graph
import org.springframework.stereotype.Component

@Component
class GraphUpdatedEventHandler : SpecificEventHandler<GraphUpdatedEvent>(GraphUpdatedEvent::class) {
	override fun handleCastEvent(game: Game<*>, event: GraphUpdatedEvent) {
		sender.send(GameGraph(game), event.graph)
	}
}

private class GameGraph(game: Game<*>): Topic<Graph<*, *>>("graph.game.${game.identifier}")